### PR TITLE
test: reorganize the TSO test and reduce the integration test time cost

### LINKED ISSUE #3838: The TSO tests is running too long
## Development Task
image
It takes about 8 minutes. This took too long.

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Resolve #3838. Reorganize the TSO test and reduce the integration test time cost.

### What is changed and how it works?

* Classify TSO tests into function tests and consistency tests.
* Only run TSO function test in a PR independently.
* Only run TSO consistency test after code is merged into the master.
* Remove some unnecessary tests.
* `make test` will only run the TSO function test.

### Check List

Tests

- Unit test
- Integration test
- Manual test
 
图片

图片

Usually, these tests will last about 11 minutes because the TSO test will be fully run in the chunk 2. Now it can be run independently, so the overall time cost will drop to about 8 minutes, which is a 27% time-saving. However, the `idc-jenkins-ci/test` still takes a long time to run. We should make some changes to the Jenkins workflow settings later.

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by disksing — APPROVED** [review]

**Review by nolouch — COMMENTED** [review]
The tso function test takes 8 minutes, it looks longer than before in the issue?

**Comment by JmPotato** [comment]
> The tso function test takes 8 minutes, it looks longer than before in the issue?

@nolouch It includes the building and dashboard downloading time.

File: tests/server/tso/allocator_test.go [hunk 1]
```diff
@@ -11,6 +11,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// +build tso_full_test tso_function_test
```
Comment (rleungx) [inline tests/server/tso/allocator_test.go:14]: What is `tso_full_test` used for?

Comment (JmPotato) [reply tests/server/tso/allocator_test.go:14]: Just in case we want to run all TSO tests by `-tags tso_full_test`.

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**